web/src/favicon.ts

```diff
@@ -19,12 +19,12 @@ function load_and_set_favicon(rendered_favicon) {
     favicon_state.image.src = favicon_state.url;
     favicon_state.image.addEventListener("load", set_favicon);
 }
-
-function set_favicon() {
-    $("#favicon").attr("href", favicon_state.url);
+function set_favicon(): void {
+    if (favicon_state !== undefined) {
+        $("#favicon").attr("href", favicon_state.url);
+    }
```

Comment:
Can you change this to log or throw throw an exception if `favicon_state` if undefined when we get here? Based on how this code is constructed, I think that's what we expect.

Reply:
I forced push a commit about this. Could you please check whether the error message is appropriate?

Reply:
I tweaked the error message and pushed back here, thanks!